src/text-field.js

```diff
@@ -30,81 +27,45 @@ class Textfield extends React.Component {
   }
 
   render(){
-
     const {
-      className,
-      type,
-      label,
-      floatingLabel,
-      error,
+      children,
       expandable,
-      button,
       icon,
-      expandableHolder
+      floating,
+      className,
+      id = this._autoId
     } = this.props;
 
-    let {
-      id
-    } = this.props;
+    const inputChildren = React.Children.map(this.props.children, function(child) {
```

Comment:
i'd probably use an arrow function here and use `this._autoId` inside this function